pkg/sources/adapter/kafkasource/adapter.go

```diff
@@ -127,19 +127,31 @@ func (a *kafkasourceAdapter) Start(ctx context.Context) error {
 		adapter: a,
 	}
 
-	wg := &sync.WaitGroup{}
-	wg.Add(1)
-
-	defer wg.Done()
-	for {
-		go func() {
+	e := make(chan error, 1)
+	go func() {
```

Comment:
So if I understand this:
1. Start function is invoked when the sources starts up
2. Spawn a new thread
2a. Inside this new thread, loop continuously and process messages as they are received
3. Block until we receive a response from the new thread of `nil` or `err`

At this point, switching from wait groups to channels doesn't yield anything as this is still "single threaded," but with the caveat that a new thread has been spun up to process the messages. Normally, it'd be best to spawn the processing in a separate thread, and provide callback handle that can be used to stop processing of all other threads.

Reply:
It is, but that's not what is implemented.

What @FranBarrera has is closer to:
```
start routine:
  fork:
    // child
    loop:
      err := dosomething()
      return err

  // parent
  block for response
  return result
```

In this case, the for loop is effectively ignored, and we're spawning a thread to do a single task, which I highly doubt is the intent.